.github/actions/bumpVersion/bumpVersion.js

```diff
@@ -1,71 +1,55 @@
-                    'Err: npm version conflict, attempting to automatically resolve',
-                    `retryCount: ${++errCount}`,
-                );
-                shouldRetry = true;
-                const {version} = JSON.parse(fs.readFileSync('./package.json'));
-                const currentPatchVersion = `v${version.slice(0, -4)}`;
-                console.log('Current patch version:', currentPatchVersion);
+const octokit = github.getOctokit(core.getInput('GITHUB_TOKEN', {required: true}));
+let semanticVersionLevel = core.getInput('SEMVER_LEVEL', {require: true});
 
-                // Get the highest build version git tag from the repo
-                console.log('Fetching tags from github...');
-                const octokit = github.getOctokit(core.getInput('GITHUB_TOKEN', {required: true}));
-                octokit.repos.listTags({
-                    owner: repoOwner,
-                    repo: repoName,
-                })
-                    .then((response) => {
-                        const tags = response.data.map(tag => tag.name);
-                        console.log('Tags: ', tags);
-                        const highestBuildNumber = Math.max(
-                            ...(tags
-                                .filter(tag => tag.startsWith(currentPatchVersion))
-                                .map(tag => tag.split('-')[1])
-                            ),
-                        );
-                        console.log('Highest build number from current patch version:', highestBuildNumber);
+// SEMVER_LEVEL defaults to BUILD
+// it actually would fall to build anyway, but I think it is better to make it explicitly
+if (!semanticVersionLevel || !Object.values(functions.semanticVersionLevels).find(v => v === semanticVersionLevel)) {
+    // eslint-disable-next-line max-len, semi
+    console.log(`Invalid input for 'SEMVER_LEVEL': ${semanticVersionLevel}, defaulting to: ${functions.semanticVersionLevels.build}`)
+    semanticVersionLevel = functions.semanticVersionLevels.build;
+}
 
-                        const newBuildNumber = `${currentPatchVersion}-${highestBuildNumber + 1}`;
-                        console.log(`Setting npm version for this PR to ${newBuildNumber}`);
-                        exec(`npm version ${newBuildNumber} -m "Update version to ${newBuildNumber}"`,
-                            // eslint-disable-next-line no-shadow
-                            (err, stdout, stderr) => {
-                                console.log(stdout);
-                                if (err) {
-                                    console.log(stderr);
-                                }
-                            });
-                    })
-                    .catch(exception => core.setFailed(exception));
-            } else {
-                core.setFailed(err);
-            }
-        }
+// eslint-disable-next-line no-shadow
+const bumpVersion = async (core, github, semanticVersionLevel) => {
```

Comment:
This may come as a bit of a surprise, but we don't use `async/await` anywhere in the Expensify.cash repo, just to maintain consistency. There is a good explanation of this in our [style guide](https://github.com/Expensify/Expensify.cash/blob/master/STYLE.md#esnext-are-we-allowed-to-use-insert-new-language-feature-why-or-why-not), which I encourage you to take a look at.

So you should use promises instead.

Reply:
I ready some of it but somehow skipped this important part, I re checked your style guide and noticed some very important information (like not using await/async and only using react components) go are a bit hidden, if you think about their importance.

I will refactor it to use promises only.